packages/local_auth/local_auth_platform_interface/lib/local_auth_platform_interface.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:local_auth_platform_interface/default_method_channel_platform.dart';
+import 'package:local_auth_platform_interface/types/auth_options.dart';
+import 'package:local_auth_platform_interface/types/biometric_type.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+/// The interface that implementations of local_auth must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `local_auth`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [LocalAuthPlatform] methods.
+abstract class LocalAuthPlatform extends PlatformInterface {
+  /// Constructs a LocalAuthPlatform.
+  LocalAuthPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static LocalAuthPlatform _instance = DefaultLocalAuthPlatform();
+
+  /// The default instance of [LocalAuthPlatform] to use.
+  ///
+  /// Defaults to [DefaultLocalAuthPlatform].
+  static LocalAuthPlatform get instance => _instance;
+
+  /// Platform-specific implementations should set this with their own
+  /// platform-specific class that extends [LocalAuthPlatform] when they
+  /// register themselves.
+  static set instance(LocalAuthPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Authenticates the user with biometrics available on the device while also
+  /// allowing the user to use device authentication - pin, pattern, passcode.
+  ///
+  /// Returns a [Future] holding true, if the user successfully authenticated,
```

Comment:
Let's just say "Returns true if [...]"; I think mentioning the Future just reads oddly, and it's a common pattern so I don't think it should cause confusion. (I know it's an existing comment, but we can take the opportunity to improve them.)